pkgs/applications/networking/instant-messengers/qq/default.nix

```diff
@@ -77,7 +78,8 @@ stdenv.mkDerivation {
       --replace "/usr/share" "$out/share"
     makeWrapper $out/opt/QQ/qq $out/bin/qq \
       --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libGL ]}" \
-      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}"
+      --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
```

Comment:
This needs a shell wrapper while `wrapGAppsHook` is a binary wrapper. Also double wrapping is bad. I'd recommend you still add `makeWrapper`, set `dontWrapGApps = true;`, use `makeShellWrapper` instead and add `''${gappsWrapperArgs[@]}` manually.

Reply:
> I'd recommend you still add `makeWrapper`, set `dontWrapGApps = true;`, use `makeShellWrapper` instead and add `''${gappsWrapperArgs[@]}` manually.

Will this breaks gtk compability? I am not sure whether `makeWrapper` could offer necessary dependencies for gtk components that helps gtk file chooser works. See https://github.com/NixOS/nixpkgs/pull/284788 for technical details

Reply:
As long as you pass those wrapper args to `makeWrapper`, they should behave in the same way.

Reply:
> As long as you pass those wrapper args to `makeWrapper`, they should behave in the same way.

It actually works, thanks! A rewritten commit is pushed into this pr.

Reply:
You didn't add `wrapGAppsHook`, is `${gappsWrapperArgs[@]}` still there?

Reply:
> You didn't add `wrapGAppsHook`, is `${gappsWrapperArgs[@]}` still there?

My bad, seems need to be removed.
From my testing, command line arguments will work out-of-box for `makeShellWrapper`, so there is no need to manually add that

Reply:
But don't you still need gtk dialogs?

Reply:
> But don't you still need gtk dialogs?

Indeed, but `makeShellWrapper` seems also cover gtk dialogs out-of-box. Don't know the exact reason, but that's how it behaves on my computer ¯\_(ツ)_/¯

Reply:
I'd suggest you kill the process, log out and log in again.

Reply:
> I'd suggest you kill the process, log out and log in again.

Tried > 5 times, logged out and logged in, still works

Reply:
Then you are editing nothing in this PR. `makeWrapper` and `makeShellWrapper` are the same thing.

Reply:
> Then you are editing nothing in this PR. `makeWrapper` and `makeShellWrapper` are the same thing.

You remind me that I find this situation on `nixpak` environment. Maybe this case is only reproducible if there is only limited env. Will try to

Reply:
Okay, after testing several times, it looks like the actually effective solution would be adding the following line:
> --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH" \

to `makeShellWrapper`. It will provide the necessary schema files.

But it's kind of weird. Shouldn't `gappsWrapperArgs[@]` already include `XDG_DATA_DIRS`? So I have tried to print out `gappsWrapperArgs[@]`, then the result is surprising:
> --prefix GIO_EXTRA_MODULES : /nix/store/4w42z3xzhsr0zxqy9w3mag42n73n1xwz-dconf-0.40.0-lib/lib/gio/modules --prefix GIO_EXTRA_MODULES : /nix/store/4w42z3xzhsr0zxqy9w3mag42n73n1xwz-dconf-0.40.0-lib/lib/gio/modules

It does not include any `XDG_DATA_DIRS` arguments.

That's seems to be why we don't have working gtk file chooser. In short, `gappsWrapperArgs` does not include `XDG_DATA_DIRS` if we refer it in `makeShellWrapper`, then the schema is missing if your env does not *implicitly* introduce `XDG_DATA_DIRS`.